run-tests.php

```diff
@@ -421,11 +421,11 @@ function main(): void
 
             switch ($switch) {
                 case 'j':
-                    $workers = substr($argv[$i], 2);
-                    if ($workers == 0 || !preg_match('/^\d+$/', $workers)) {
-                        error("'$workers' is not a valid number of workers, try e.g. -j16 for 16 workers");
+                    $workers_raw = substr($argv[$i], strlen("-j"));
+                    $workers = try_string_to_int($workers_raw, 1);
+                    if ($workers === null) {
+                        error("'{$workers_raw}' is not a valid number of workers, try e.g. -j16 for 16 workers");
```

Comment:
imo using`{}` is less error-prone, so i pretty much always use {} - take for example concatenating 3 variables with `_`, people who habitually do not use `{}` may write it as
```php
$str = "$v1_$v2_$v3";
```
which is wrong: it tries to concatenate the variable `$v1_` and `$v2_` and `$v3_` with no separator, instead of concatenating `$v1` and `$v2` and `$v3` with `_` , but it may be difficult to spot the error in code review. people who habitually use `{}` would write it as
```php
$str = "{$v1}_{$v2}_{$v3}";
```
which would actually work.

Are you sure I should really remove the `{}` ?